python/cudf_polars/cudf_polars/dsl/expressions/unary.py

```diff
@@ -317,11 +317,13 @@ def do_evaluate(
                 evaluated = self.children[1].evaluate(df, context=context)
                 arg = evaluated.obj_scalar if evaluated.is_scalar else evaluated.obj
             if isinstance(arg, plc.Scalar) and dtypes.can_cast(
-                column.obj.type(), arg.type()
+                column.dtype.plc, arg.type()
             ):  # pragma: no cover
-                arg = plc.unary.cast(
-                    plc.Column.from_scalar(arg, 1), column.obj.type()
-                ).to_scalar()
+                arg = (
+                    Column(plc.Column.from_scalar(arg, 1), dtype=DataType(arg.type()))
```

Comment:
Is this call the only place why we need to have `DataType` accept more than just a Polars DataType?

Reply:
I don't think we know the corresponding polars dtype of `arg`.

Reply:
Oh wait I think we do. We can get it from `Literal` (the type before evaluation).

Reply:
I'll try and remove the `_to_polars` function